### PR TITLE
OS History API - Swagger

### PR DESCRIPTION
**Pull request by sds6**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-libraries-for-java

**Comment by sarangan12** [comment]
@sds6 The contributor license agreement has not been signed yet. Could you join the appropriate org to resolve the issue?

**Comment by sds6** [comment]
@sarangan12 signed.

**Comment by sarangan12** [comment]
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/357213506
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/357213505

Please take a look at these errors and try to fix them.

**Comment by sarangan12** [comment]
@sds6 Any updates????

**Comment by sds6** [comment]
Will have an update early next week.

**Comment by sds6** [comment]
@sarangan12 All the errors I see are for x-ms-examples.  Compute is adding them gradually for all APIs over the next few months. Similar PR #2589 was also approved ignoring x-ms-examples. Adding in @hyonholee to chime in.

**Review by sarangan12 — APPROVED** [review]

**Comment by sarangan12** [comment]
@sds6 Take a look at this: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/357213505 These errors are not related to x-ms-examples. These errors must be fixed before approval

**Review by sarangan12 — CHANGES_REQUESTED** [review]
Please see the link https://travis-ci.org/Azure/azure-rest-api-specs/jobs/357213505 These are not x-ms-examples errors. They must be fixed before approval

**Comment by sarangan12** [comment]
@sds6 The recent changes have introduced more new errors. Please check the travis CI.

**Comment by sds6** [comment]
@sarangan12 Those changes are not related to the push I made.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json [hunk 1]
```diff
@@ -7121,6 +7163,145 @@
...
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "Start time of the upgrade."
+        },
+        "endTime": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "Start time of the upgrade."
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: "End time of the upgrade"

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json [hunk 2]
```diff
@@ -7121,6 +7163,145 @@
...
+        },
+        "error": {
+          "$ref": "#/definitions/ApiError",
+          "readOnly": true,
+          "description": "Error Details for this upgrade if there are any."
+        },
+        "startedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Error Details for this upgrade if there are any.",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: Update the description

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json [hunk 3]
```diff
@@ -7121,6 +7163,145 @@
...
+          "readOnly": true,
+          "description": "Information about the properties of the upgrade operation."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: "readOnly": true,

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json [hunk 4]
```diff
@@ -7121,6 +7163,145 @@
...
+          "description": "Resource location"
+        }
+      },
+        "description": "Virtual Machine Scale Set OS Upgrade History operation response."
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: NIT: fix indent

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**